usl_pipeline/cloud_functions/main.py

```diff
@@ -1027,6 +1028,81 @@ def _direction_to_cosine(degrees):
     return dataset
 
 
+def _compute_solar_time_components(dataset: xarray.Dataset) -> xarray.Dataset:
+    # Solar Time Computation for every Longitude of the City
+    # Solar Time Sine and Cosine Derivation from there
+    if all(var in dataset.keys() for var in ["XLONG_M", "XLAT_M", "Times"]):
+        # Extract longitude, latitude, and time variables from dataset
+        longitudes = dataset["XLONG_M"][0, :, :]
+        latitudes = dataset["XLAT_M"][0, :, :]
+        times = dataset["Times"]
+
+        # Convert time variable to datetime objects
+        times = xarray.DataArray(
+            [
+                # Check if the value is a Unix timestamp (numeric type)
+                (
+                    numpy.datetime64(
+                        datetime.datetime.utcfromtimestamp(t).strftime(
+                            "%Y-%m-%dT%H:%M:%S"
+                        )
+                    )
+                    if isinstance(t, (int, float))
+                    else numpy.datetime64("".join(t.astype(str)).replace("_", "T"))
+                )
+                for t in times.values
+            ]
+        )
+
+        # Extract hours and minutes from the datetime objects
+        utc_hours_minutes = (
+            times.astype("datetime64[h]").astype(int) % 24
+            + times.astype("datetime64[m]").astype(int) % 60 / 60
+        )
+
+        # Define the function to calculate solar time
+        def calculate_solar_time(utc_time, longitude):
+            return (utc_time + longitude / 15) % 24
```

Comment:
what if we have negative values of longitude (west of the prime meridian)?
we may adjust as 
return ((utc_time + longitude / 15) % 24 + 24) % 24

Reply:
For negative longitudes like NYC the code will take care of it. Eg, if WRF variable XLONG_M = -75, then local time would be UTC-(5) hrs. What I am concerned about is the sin and cos solar time values that I was getting in my code and the values that the unit tests in this one is giving